receiver/dotnetdiagnosticsreceiver/network/reader.go

```diff
@@ -0,0 +1,247 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+	"unicode/utf16"
+)
+
+// MultiReader provides an abstraction to make reading from the diagnostics
+// stream easier for callers.
+type MultiReader interface {
+	Pos() int
+	Align() error
+	Seek(i int) error
+	ReadUTF16() (string, error)
+	ReadByte() (byte, error)
+	AssertNextByteEquals(byte) error
+	ReadCompressedUInt64() (uint64, error)
+	ReadCompressedInt64() (int64, error)
+	ReadCompressedUInt32() (uint32, error)
+	ReadCompressedInt32() (int32, error)
+	Read(interface{}) error
+	ReadASCII(int) (string, error)
+	Reset()
```

Comment:
Please add comments. `Align`, `Read`, `ReadASCII` or `Reset` for example not obvious.

Reply:
Sure, done.